crates/sui-indexer-alt-framework/src/handlers/cp_mapping.rs

```diff
@@ -20,8 +20,61 @@ pub struct StoredCpMapping {
     pub epoch: i64,
 }
 
+pub struct PrunableRange {
+    from: StoredCpMapping,
+    to: StoredCpMapping,
+}
+
 pub struct CpMapping;
 
+impl PrunableRange {
+    /// Gets the tx and epoch mappings for both the start and end checkpoints.
+    ///
+    /// The values are expected to exist since the cp_mapping table must have enough information to
+    /// encompass the retention of other tables.
+    pub async fn get_range(
+        conn: &mut Connection<'_>,
+        from_cp: u64,
+        to_cp: u64,
+    ) -> QueryResult<PrunableRange> {
+        let results = cp_mapping::table
+            .select(StoredCpMapping::as_select())
+            .filter(cp_mapping::cp_sequence_number.eq_any([from_cp as i64, to_cp as i64]))
+            .order(cp_mapping::cp_sequence_number.asc())
+            .load::<StoredCpMapping>(conn)
+            .await?;
+
+        match results.as_slice() {
+            [first, .., last] => Ok(PrunableRange {
+                from: first.clone(),
+                to: last.clone(),
+            }),
+            _ => Err(diesel::result::Error::NotFound),
+        }
+    }
+
+    /// Inclusive start and exclusive end range of prunable checkpoints.
```

Comment:
Whether or not the end is exclusive depends on whether `to` field is exclusive, which isn't a given?
Maybe worth to rename the field and make it clear in the PrunableRange doc/constructor as well.

Reply:
That's true, but I don't think users should instantiate the `CheckpointRange` themselves; rather it would get instantiated by the pruner task, and its checkpoints `from` and `to` are expected to come from `watermark.next_chunk`. Since the two values come from `pruner_hi`, and the min of `(from + size)` or `reader_lo`, it should be fair to say that `to` would consequently be exclusive

Reply:
Right, so the suggestion would be to make that explicit in the definition of `CheckpointRange`.

Reply:
In the codebase, the convention seems to be that `from` is inclusive and `to` is exclusive, and If `to` needs to be inclusive, we add "inclusive" to the name. One example is `checkpoint_hi_inclusive` in `CommitterWatermark`. So I think leaving it as `to` is good enough.